packages/flutter_tools/lib/src/web/file_generators/flutter_js.dart

```diff
@@ -8,24 +8,141 @@
 /// flutter.js should be completely static, so **do not use any parameter or
 /// environment variable to generate this file**.
 String generateFlutterJsFile() {
-  return '''
+  return r'''
 // Copyright 2014 The Flutter Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-/**
- * This script installs service_worker.js to provide PWA functionality to
- *     application. For more information, see:
- *     https://developers.google.com/web/fundamentals/primers/service-workers
- */
-
 if (!_flutter) {
   var _flutter = {};
 }
 _flutter.loader = null;
 
 (function() {
   "use strict";
+  /**
+   * Wraps `promise` in a timeout of the given `duration` in ms.
+   *
+   * Resolves/rejects with whatever the original `promises` does, or rejects
+   * if `promise` takes longer to complete than `duration`. In that case,
+   * `debugName` is used to compose a legible error message.
+   *
+   * If `duration` is <= 0, the original `promise` is returned unchanged.
+   * @param {Promise} promise
+   * @param {number} duration
+   * @param {string} debugName
+   * @returns {Promise} a wrapped promise.
+   */
+  async function timeout(promise, duration, debugName) {
+    if (duration <= 0) {
+      return promise;
+    }
+    let _timeoutId;
```

Comment:
Is there a reason we're underscore prefixing these locals?

Reply:
Haha absolutely none, it's my Dartvascript showing :P